src/main/java/com/newts/newtapp/api/application/conversation/GetUserList.java

```diff
@@ -0,0 +1,61 @@
+package com.newts.newtapp.api.application.conversation;
+
+import com.newts.newtapp.api.ConversationRepository;
+import com.newts.newtapp.api.UserRepository;
+import com.newts.newtapp.api.application.*;
+import com.newts.newtapp.api.errors.ConversationNotFound;
+import com.newts.newtapp.api.errors.EmptyConversation;
+import com.newts.newtapp.api.errors.UserNotFound;
+import com.newts.newtapp.entities.Conversation;
+import com.newts.newtapp.entities.User;
+
+import java.util.ArrayList;
+
+/**
+ * ConversationInteractor that returns the list of user profiles of the conversation.
+ * RequestModel must provide the conversation id
+ */
+public class GetUserList extends ConversationInteractor<ArrayList<UserProfile>, Exception> {
+    private UserRepository userRepository;
+    private ConversationRepository conversationRepository;
+
+    /**
+     * Initialize a new GetUserList interactor with given ConversationRepository.
+     * @param userRepository UserRepository to access user data by
+     * @param conversationRepository ConversationRepository to access conversation data by
+     */
+    public GetUserList(ConversationRepository conversationRepository, UserRepository userRepository) {
+        super(conversationRepository, userRepository); }
+
+    /**
+     * Returns the list of user profiles of the conversation.
+     * @param  request              a request stored as a RequestModel
+     * @return userProfiles         a list of user profiles of the conversation
+     * @throws ConversationNotFound if conversation does not exist
+     * @throws EmptyConversation    if the conversation has no users
+     * @throws UserNotFound         if user does not exist
+     */
+    @Override
+    public ArrayList<UserProfile> request(RequestModel request) throws ConversationNotFound, EmptyConversation,
+            UserNotFound {
+        int conversationId = (int) request.get(RequestField.CONVERSATION_ID);
+
+        Conversation conversation = conversationRepository.findById(conversationId).orElseThrow(ConversationNotFound::new);
+
+        // Check if conversation is empty
+        if (conversation.getNumUsers() == 0) {
```

Comment:
Can get rid of this case. This case shouldn't happen and if it does, an empty list is fine.